### PR TITLE
Close the UrlLoaderProxy on error.

### LINKED ISSUE #1680: fetchUrl does not call close() on exceptions
In sky/packages/sky/lib/src/services/fetch.dart, loader.close() is not called if an exception is thrown. I would propose that loader.close() should be called in a try/finally block.

### PR DESCRIPTION
**Pull request by jimbeveridge**
Fixes #1680.

### CODE REVIEW

File: sky/packages/sky/lib/src/services/fetch.dart [hunk 1]
```diff
@@ -53,7 +54,7 @@ Future<UrlResponse> fetchUrl(String relativeUrl) async {
   UrlRequest request = new UrlRequest()
     ..url = url
     ..autoFollowRedirects = true;
-  return await fetch(request);
+  return fetch(request);
```
Comment (abarth) [inline sky/packages/sky/lib/src/services/fetch.dart]: I don't think this change is correct...

Comment (abarth) [inline sky/packages/sky/lib/src/services/fetch.dart]: I think you want to remove the `async` keyword from the declaration to make it right.

Comment (jimbeveridge) [inline sky/packages/sky/lib/src/services/fetch.dart]: fetch() and Future<UrlResponse> both return a Future<UrlResponse>, so this return just passes it through. The only reason you need an await is if you were doing some additional processing before returning. @johnmccutchan, could you please confirm or deny this?

Comment (johnmccutchan) [inline sky/packages/sky/lib/src/services/fetch.dart]: If this CL was applied, fetchUrl would work with and without the async modifier. Now that fetchUrl doesn't use await, it is more efficient to drop the async.

Comment (jimbeveridge) [inline sky/packages/sky/lib/src/services/fetch.dart]: Done. Removed async from fetchUrl().

**Comment by abarth** [comment]
LGTM

**Comment by jimbeveridge** [comment]
I do not have write access to this repo, so I cannot merge the pull request.

**Comment by abarth** [comment]
I've invited you.

**Pull request merged**